resources/lib/api.py

```diff
@@ -178,38 +160,30 @@ def get_episode_id(showid, show_season, show_episode):
             tvshowid=int(showid),
         ))
 
-        if not result.get('result', {}).get('episodes'):
-            return 0
-
         episodeid = 0
-        for episode in result.get('result').get('episodes'):
+        for episode in result.get('result').get('episodes', []):
             if episode.get('episodeid') and episode.get('season') == show_season and episode.get('episode') == show_episode:
                 episodeid = episode.get('episodeid')
 
         return episodeid
 
     def find_next_episode(self, result, current_file, include_watched, current_episode_id):
-        position = 1
-        for episode in result.get('result').get('episodes'):
+        found_match = False
+        episodes = result.get('result', {}).get('episodes', [])
+        for episode in episodes:
             # Find position of current episode
             if current_episode_id == episode.get('episodeid'):
-                # Found a match so get out of here
-                break
-            position += 1
-
-        # Check if it may be a multi-part episode
-        while result.get('result').get('episodes')[position].get('file') == current_file:
-            position += 1
-
-        # Skip already watched episodes?
-        while not include_watched and result.get('result').get('episodes')[position].get('playcount') > 1:
-            position += 1
-
-        try:
-            episode = result.get('result').get('episodes')[position]
-        except (IndexError, KeyError) as exc:
```

Comment:
The IndexError and KeyError at this stage is useless as we may already have accessed it.
This is what needed fixing.